src/win.rs

```diff
@@ -117,26 +118,37 @@ impl Clone for HandleRef {
     }
 }
 
+/// Ensures the handle is never null by converting null values to the
+/// pseudo-handle `INVALID_HANDLE_VALUE`.
+fn nonnull_handle<T: AsRawHandle>(stdio: T) -> RawHandle {
+    let handle = stdio.as_raw_handle();
+    if handle.is_null() {
+        INVALID_HANDLE_VALUE
+    } else {
+        handle
+    }
+}
```

Comment:
My guess is that trying to write to such a handle will just result in nothing happening, right? And I think that's what the behavior was before this change, so I guess this doesn't make anything worse.

Reply:
It should be an `Err` when used rather than a panic when created, which was the status quo before the standard library added the panic.